flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/SinkITCase.java

```diff
@@ -256,6 +275,95 @@ public void writerAndGlobalCommitterExecuteInBatchMode() throws Exception {
                 containsInAnyOrder(EXPECTED_GLOBAL_COMMITTED_DATA_IN_BATCH_MODE.toArray()));
     }
 
+    @Test
+    public void testMetrics() throws Exception {
```

Comment:
The code for the source metric tests looks very similar can you try to generify this a bit?

Reply:
The are in completely unrelated packages and I hate to generify it to something completely independent. Maybe we should embed it into the connector testing framework when it arrives and is beefed for sinks.